File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uset;
        for(int n : nums){
            uset.insert(n);
        }

        int ans = 0;
        for(int n : nums){
            if(uset.find(n-1) == uset.end()){
                int count = 1;
                while(uset.find(n + 1) != uset.end()){
                    n++;
                    count ++;
                }
                ans = max(ans, count);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-7.java
class Solution {
    public int trap(int[] h) {
        int n = h.length;
        int lm = h[0];
        int rm = h[n-1];

        int i = 0, j = n - 1;
        int ans = 0;
        while(i <= j) {
            lm = Math.max(lm, h[i]);
            rm = Math.max(rm, h[j]);

            if (lm < rm) {
                ans += Math.max(0, Math.min(lm, rm) - h[i]);
                i++;
            } else {
                ans += Math.max(0, Math.min(lm, rm) - h[j]);
                j--;
            }
        }

        return ans;
    }
}